source/base/ByteBuffer.ooc

```diff
@@ -28,26 +28,29 @@ ByteBuffer: class {
 	size ::= this _size
 	_referenceCount: ReferenceCounter
 	referenceCount ::= this _referenceCount
-	init: func (=_pointer, =_size) {
+	_owner: Bool
+	_forceFree := false
+
+	init: func (=_pointer, =_size, owner := false) {
 		this _referenceCount = ReferenceCounter new(this)
+		this _owner = owner
 	}
-	__destroy__: func {
+	free: override func {
 		if (this _referenceCount != null)
 			this _referenceCount free()
 		this _referenceCount = null
-		gc_free(this _pointer)
+		if (this _owner)
+			gc_free(this _pointer)
 		this _pointer = null
 		super()
 	}
-	zero: func ~whole {
-		memset(_pointer, 0, _size)
-	}
-	zero: func ~range (offset, length: Int) {
-		memset(_pointer + offset, 0, length)
-	}
-	slice: func (offset, size: Int) -> This {
-		_SlicedByteBuffer new(this, offset, size)
+	forceFree: func {
```

Comment:
I still don't like the fact that we apparently need this. No one who uses a `ByteBuffer` should have to keep track of how it frees itself, it should be abstracted and kept inside the class. Can we find another way?